src/averageAges.js

```diff
@@ -37,7 +41,18 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenIsMotherWithAge = people.map(person => ({
+    ...person,
+    isMother: people.some(one => one.mother === person.name),
+    age: person.died - person.born,
+  })).filter(person => person.sex === 'f');
```

Comment:
serach for woman's children and check if she is a mother only if you have `withChildren` flag, it helps optimise performance